### PR TITLE
CLI GitHub Action

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds a simple GitHub CI action from the price-calculator repo. The action will run the test and compile the codebase on each push

### CODE REVIEW

**Comment by arielmelendez** [comment]
Any explanation for changes in the CLI layer?

**Comment by fedellen** [comment]
> Any explanation for changes in the CLI layer?

Well, the GitHub action was initially failing due to that parameter I removed: https://github.com/ardriveapp/ardrive-cli/actions/runs/1279875121

This commit made the action work: 19634cd

`yarn test` and self-testing still functioned fine after removal, so I asked Mati if removing that code would be fine in his commander API. He confirmed that it should be okay.

So I didn't really dig too much further into it after this workaround. Not the best explanation, and I'm still not sure why the tests were failing on GH and not for me locallly

File: package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
 		"coverage": "nyc --reporter text mocha",
 		"typecheck": "tsc --noemit",
 		"build": "yarn clean && tsc",
+		"ci": "yarn test && yarn build",
```
[the remark this answers is not in the document]
Comment (fedellen) [reply package.json]: I'm not sure whether it matters, but added the change in case it does 👍

**Comment by matibat** [comment]
I think this mocha env is re-using the same module instances. That's why it failed the 1st time.

I wonder if there's a way of running a test after another has run, because it fails just after parse() is called:
```
  CLICommand class
    ✔ Calls the library API function once when a command is set
    ✔ The library parses the given argv
    ✔ CLICommand won't allow to declare a command after parsed


  CommonContext class
    1) Actually reads the value from argv
    2) Boolean parameter false
    3) Boolean parameter true
    4) Array parameter
    ✔ Required parameter throws if missing
```

But I'm just fine with removing the _doneSettingCommands parameter, that's easier

**Review by matibat — APPROVED** [review]
I would say it's good

**Pull request merged**